api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/ListOfPatientsCurrentlyOnArtWithoutTbScreeningDataset.java

```diff
@@ -0,0 +1,237 @@
+import org.openmrs.module.eptsreports.reporting.library.cohorts.*;
+import org.openmrs.module.eptsreports.reporting.library.queries.TxtbDenominatorQueries;
+import org.openmrs.module.reporting.data.DataDefinition;
+import org.openmrs.module.reporting.data.converter.DataConverter;
+import org.openmrs.module.reporting.data.converter.ObjectFormatter;
+import org.openmrs.module.reporting.data.patient.definition.ConvertedPatientDataDefinition;
+import org.openmrs.module.reporting.data.patient.definition.PatientIdentifierDataDefinition;
+import org.openmrs.module.reporting.data.person.definition.ConvertedPersonDataDefinition;
+import org.openmrs.module.reporting.data.person.definition.GenderDataDefinition;
+import org.openmrs.module.reporting.data.person.definition.PersonAttributeDataDefinition;
+import org.openmrs.module.reporting.data.person.definition.PreferredNameDataDefinition;
+import org.openmrs.module.reporting.dataset.definition.DataSetDefinition;
+import org.openmrs.module.reporting.dataset.definition.PatientDataSetDefinition;
+import org.openmrs.module.reporting.evaluation.parameter.Parameter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ListOfPatientsCurrentlyOnArtWithoutTbScreeningDataset extends BaseDataSet {
+
+  @Autowired
+  private ListOfPatientsCurrentlyOnArtWithoutTbScreeningCohortQueries
+      listOfPatientsCurrentlyOnArtWithoutTbScreeningCohortQueries;
+
+  @Autowired private ListOfPatientsArtCohortCohortQueries listOfPatientsArtCohortCohortQueries;
+
+  @Autowired private TPTListOfPatientsEligibleDataSet tptListOfPatientsEligibleDataSet;
+
+  @Autowired
+  private ListOfPatientsDefaultersOrIITCohortQueries listOfPatientsDefaultersOrIITCohortQueries;
+
+  @Autowired private TPTInitiationDataDefinitionQueries tptInitiationDataDefinitionQueries;
+
+  @Autowired
+  private ListOfPatientsEligibleForVLDataDefinitionQueries
+      listOfPatientsEligibleForVLDataDefinitionQueries;
+
+  @Autowired private TxtbDenominatorQueries txtbDenominatorQueries;
+
+  @Autowired private HivMetadata hivMetadata;
+
+  public DataSetDefinition constructListOfPatientsDataset() {
+    PatientDataSetDefinition patientDefinition = new PatientDataSetDefinition();
+    patientDefinition.setName("List of patients currently no ART without TB Screening");
+    patientDefinition.setParameters(getParameters());
+
+    patientDefinition.addRowFilter(
+        listOfPatientsCurrentlyOnArtWithoutTbScreeningCohortQueries
+            .getPatientsCurrentlyOnArtWithoutTbScreening(),
+        "endDate=${endDate},location=${location}");
+
+    PatientIdentifierType identifierType =
+        Context.getPatientService()
+            .getPatientIdentifierTypeByUuid("e2b966d0-1d5f-11e0-b929-000c29ad1d07");
+
+    DataConverter identifierFormatter = new ObjectFormatter("{identifier}");
+
+    DataDefinition identifierDef =
```

Comment:
Not used in the dataset, was it supposed to be the NID?

Reply:
if NOT going to be used, please consider removing

Reply:
Removed